### PR TITLE
Add user notifications and emails

### PR DESCRIPTION
**Pull request by nrotstan**
* Allow users to subscribe to various types of in-app notifications
* Allow users to request an email when notifications are delivered
* Users can choose immediate, digest, or no email per notification type
* Support notifications for comment mentions and task-review updates
* Add `email` column to users table to track user email addresses
* Add SMTP support for transmitting emails via play-mailer
* Add scheduler job for processing immediate notification emails
* Add scheduler job for processing digest notification emails
* Add new mailer settings block to application.conf
* Update README with section on SMTP configuration
* To aid comment mentions, make lookups by OSM username case insensitive

### CODE REVIEW

File: app/org/maproulette/models/dal/TaskDAL.scala [hunk 1]
```diff
@@ -1655,6 +1659,59 @@ class TaskDAL @Inject()(override val db: Database,
     }
   }
 
+  def generateMentionNotifications(fromUser:User, comment:Comment, taskId:Long)(implicit c:Connection=null) = {
```
Comment (mgcuthbert) [inline app/org/maproulette/models/dal/TaskDAL.scala]: So this particular class is getting rather large and taking on a log of responsibility that should be moved. My plan is to refactor this class. However new functionality that could be abstracted out should be. The notifications are definitely something that should be abstracted out into a separate class. The DAL class should really only be related to access the object data from the data storage, and this has become very muddy over the years. So if possible can we move this to a separate service.

File: app/org/maproulette/models/dal/TaskDAL.scala [hunk 2]
```diff
@@ -1655,6 +1659,59 @@ class TaskDAL @Inject()(override val db: Database,
...
+                extra=Some(comment.comment),
+              ))
+            case None => None
+          }
+        }
+      case None => throw new NotFoundException(s"No task [$taskId] found")
+    }
+  }
+
+  def generateReviewNotification(user: User, forUserId: Int, reviewStatus: Int, task: Task, comment: Option[Comment])(implicit c:Connection=null) = {
```
Comment (mgcuthbert) [inline app/org/maproulette/models/dal/TaskDAL.scala]: Same as above comment.

File: app/org/maproulette/models/dal/TaskDAL.scala (hunk 1 above)
Comment (nrotstan) [reply app/org/maproulette/models/dal/TaskDAL.scala]: I've extracted the notification-related code into new NotificationDAL and NotificationController classes, unburdening the TaskDAL, UserDAL, and UserController classes.

**Comment by mgcuthbert** [comment]
@nrotstan Just need to resolve the conflicts so that we can merge this.

**Comment by nrotstan** [comment]
Thanks, @mgcuthbert, conflict resolved.

**Comment by mgcuthbert** [comment]
I am sorry @nrotstan I merged a couple other PR's before looking at this one again, and it caused another conflict that needs resolution. My apologies.

**Comment by nrotstan** [comment]
No worries, @mgcuthbert, conflicts resolved.

**Pull request merged**